app/repos/cdo.js

```diff
@@ -468,20 +462,35 @@ const saveCdoTaskList = async (cdoTaskList, transaction) => {
   const cdoDao = await getCdo(cdoTaskList.cdoSummary.indexNumber)
 
   for (const update of updates.exemption) {
-    const mapping = updateMappings[update.key].split('.')
-    const field = mapping.pop()
-    const [, subModel] = mapping
-    const model = cdoDao[subModel]
+    const mappingArray = updateMappings[update.key].split('.')
 
-    model[field] = update.value
+    if (mappingArray[1] === 'registration') {
+      const [,, field] = mappingArray
+      const registration = cdoDao.registration
+      registration[field] = update.value
 
-    await model.save({ transaction })
+      await registration.save({ transaction })
+    } else if (mappingArray[1] === 'insurance') {
```

Comment:
With having these 'if' conditions (not that you can easily get around them) we are losing the ability of having a generic process. i.e. we are almost doing separate methods for each task. So the code would have been possibly much easier if we'd had a simple 'case' statement calling the appropriate methods to save each task's data from the route. I'm not suggesting changing anything, I'm just processing out loud